### ISSUE TITLE
Inconsistent overload with `Concatenate`

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#3113]
**Describe the bug**
Possibly a false positive when using `Concatenate` in a nested Callable with overloads.

**To Reproduce**
```py
from typing import Callable, TypeVar, overload
from typing_extensions import Concatenate, ParamSpec

class A: ...

_T = TypeVar("_T", bound=A)
_R = TypeVar("_R")
_P = ParamSpec("_P")


@overload
def error_decorator(
    error_codes: None = None,
) -> Callable[[Callable[Concatenate[_T, _P], _R]], Callable[Concatenate[_T, _P], _R]]:
    ...

@overload
def error_decorator(
    error_codes: list[str],
) -> Callable[
    [Callable[Concatenate[_T, _P], _R]], Callable[Concatenate[_T, _P], _R | None]
]:
    ...

def error_decorator(  # <-- error --
    error_codes: list[str] | None = None,
) -> Callable[
    [Callable[Concatenate[_T, _P], _R]], Callable[Concatenate[_T, _P], _R | None]
]:
    """Filter specific errors and raise custom exception for remaining once."""

    def decorator(
        func: Callable[Concatenate[_T, _P], _R]
    ) -> Callable[Concatenate[_T, _P], _R | None]:

        def wrapper(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> _R | None:
            try:
                return func(self, *args, **kwargs)
            except Exception as ex:
                if error_codes is not None:
                    return None
                raise Exception("Custom exception") from ex

        return wrapper
    return decorator
```
```
  test.py:25:5 - error: Overloaded function implementation is not consistent with signature of overload 1
    Function return type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator)" is incompatible with type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)"
      Type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator)" cannot be assigned to type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)"
        Parameter 1: type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" cannot be assigned to type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator"
          Type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" cannot be assigned to type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" (reportGeneralTypeIssues)
  test.py:25:5 - error: Overloaded function implementation is not consistent with signature of overload 2
    Function return type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)" is incompatible with type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)"
      Type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)" cannot be assigned to type "((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator) -> ((_T@error_decorator, **_P@error_decorator) -> _R@error_decorator | None)"
        Parameter 1: type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" cannot be assigned to type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator"
          Type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" cannot be assigned to type "(_T@error_decorator, **_P@error_decorator) -> _R@error_decorator" (reportGeneralTypeIssues)
```

**Expected behavior**
_No error_

**VS Code extension or command-line**
Command-line: `v1.1.224`

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in 1.1.226, which I just published. It will also be included in the next release of pylance.
